worldedit-core/src/main/java/com/fastasyncworldedit/core/queue/implementation/SingleThreadQueueExtent.java

```diff
@@ -52,6 +54,7 @@ public class SingleThreadQueueExtent extends ExtentBatchProcessorHolder implemen
     // private static final ConcurrentLinkedQueue<IChunk> CHUNK_POOL = new ConcurrentLinkedQueue<>();
     // Chunks currently being queued / worked on
     private final Long2ObjectLinkedOpenHashMap<IQueueChunk> chunks = new Long2ObjectLinkedOpenHashMap<>();
+    private final Queue<Long> chunkOrder = new ArrayDeque<>(Settings.settings().QUEUE.TARGET_SIZE);
```

Comment:
Why do we need this? Shouldn't the linked hashmap already ensure that we remove oldest first?

Reply:
Ah yeah, I had a moment